### PR TITLE
feature/MYST-376: display error strip when proposal list is empty

### PR DESCRIPTION
**Pull request by shroomist**
and if proposal fetching throws

### CODE REVIEW

File: src/renderer/components/CountrySelect.vue [hunk 1]
```diff
@@ -83,9 +83,9 @@
       },
       async fetchCountries () {
         this.countriesAreLoading = true
-
         try {
           const response = await tequilapi.proposal.list()
+          if (response.proposals.length < 1) throw new Error('Empty proposal list received')
```
Comment (Waldz) [inline src/renderer/components/CountrySelect.vue]: Be consistent then formatting code

File: test/unit/specs/components/CountrySelect.spec.js [hunk 2]
```diff
@@ -46,39 +49,91 @@ const tequilapiMock = () => {
...
+        [type.SHOW_ERROR] (state, error) {
+          state.errorMessage = error.message
+        }
+      }
+    }
+    beforeEach(() => {
+      store = new Vuex.Store(storeObj)
+    })
+
+    it('commits \'' + messages.countriesLoadingFailed + '\' when empty proposal list is received', async () => {
```
Comment (Waldz) [inline test/unit/specs/components/CountrySelect.spec.js]: Strange escaping

Comment (ignasbernotas) [inline test/unit/specs/components/CountrySelect.spec.js]: You can use string literals here `commits ${messages.countriesLoadingFailed} when empty proposal list is received`

File: src/renderer/components/CountrySelect.vue (hunk 1 above)
Comment (Waldz) [reply src/renderer/components/CountrySelect.vue]: Empty list is not error.
And if we need this logic so much it should be in on Mysterion side

File: test/unit/specs/components/CountrySelect.spec.js [hunk 3]
```diff
@@ -46,39 +49,91 @@ const tequilapiMock = () => {
...
+      expect(wrapper.vm.$store.state.errorMessage).to.eql(messages.countriesLoadingFailed)
+    })
 
-    expect(wrapper.vm.countriesList).to.have.lengthOf(4)
-    expect(wrapper.findAll('.multiselect__option-title')).to.have.lengthOf(4)
-    expect(wrapper.text()).to.contain('Lithuania')
-    expect(wrapper.text()).to.contain('United Kingdom')
-    expect(wrapper.text()).to.contain('0x3')
-    expect(wrapper.text()).to.contain('N/A')
+    it('commits \'' + messages.countriesLoadingFailed + '\' when proposal listing throws', async () => {
```
Comment (ignasbernotas) [inline test/unit/specs/components/CountrySelect.spec.js]: string literals here as well :)

File: test/unit/specs/components/CountrySelect.spec.js [hunk 4]
```diff
@@ -46,39 +49,91 @@ const tequilapiMock = () => {
...
+      CountrySelect.__Rewire__('tequilAPI', tequilapiMockThrows)
```
Comment (Waldz) [inline test/unit/specs/components/CountrySelect.spec.js]: Would be good to rebase from `feature/MYST-376-update-proposals`

File: src/renderer/components/CountrySelect.vue (hunk 1 above)
Comment (shroomist) [reply src/renderer/components/CountrySelect.vue]: not throwing Error anymore, but displaying Error strip with text
`'No countries available'`

**Pull request merged**